targets/TARGET_STM/TARGET_STM32F4/gpio_irq_api.c

```diff
@@ -313,7 +313,7 @@ void gpio_irq_set(gpio_irq_t *obj, gpio_irq_event event, uint32_t enable)
                 mode = STM_MODE_IT_RISING;
                 obj->event = EDGE_RISE;
             } else { // NONE or FALL
-                mode = STM_MODE_IT_EVT_RESET;
+                mode = STM_MODE_IT_RISING;
```

Comment:
STM_MODE_IT_EVT_RESET is not a defined mode for ST HAL. We have to set a mode among possible values.